mne_bids/read.py

```diff
@@ -331,18 +331,21 @@ def read_raw_bids(bids_basename, bids_root, modality=None, extra_params=None,
     sub = bids_basename.subject
     ses = bids_basename.session
 
+    # set root, infer the modality and
+    # then set it to the modality and suffix of the BIDSPath
+    bids_basename.update(root=bids_root)
     if modality is None:
         modality = _infer_modality(bids_root=bids_root,
                                    sub=sub, ses=ses)
```

Comment:
We wanted to stop doing that, no? Or is it still useful? I'm undecided!

Reply:
Ok I had a look at the code, so we're trying to infer `modality` from the suffix for MEG, EEG, iEEG. Which is … probably fine? But it will lead to an inconsistency when using an anatomy suffix such as `T1w`, right? Shouldn't we add those there, too?